### DISCUSSION TITLE
whether the strategy of Karmada updating the workloads is RollingUpdate or Recreate?

### DISCUSSION THREAD

**Discussion by DAJINZI01** [karmada-io/karmada#5900] [General]
Suppose there is the following scenario: initially, there is a workload with 4 replicas in Cluster 1. By editing the PropagationPolicy, the propagation scope is changed to Cluster 2. Would the update of this workload be smooth? In other words, would the replicas in Cluster 1 only be deleted after the workload becomes healthy in Cluster 2, ensuring no interruption in traffic during the update process?

**Comment by RainbowMango** [comment]
Why not change the PropagationPolicy and add cluster 2 along with cluster 1? And then remove cluster 1 from it once the workload has been provisioned in cluster 2.

Same question as https://cloud-native.slack.com/archives/C02MUF8QXUN/p1732233542466079.

**Reply by DAJINZI01** [reply]
Your meaning is:

Karmada's scheduling does not consider pod graceful migration; it simply adjusts the replica counts across different clusters?
The strategy you described involves modifying the PropagationPolicy to include cluster 2, then scaling up the corresponding workload replicas from 4 to 8, and then scheduling the new replicas to cluster 2 and ensuring they are healthy before removing cluster 1 from the PropagationPolicy.
Why doesn't Karmada's scheduling consider smooth and graceful operations?

PS: The provided link is not accessible within the company.
